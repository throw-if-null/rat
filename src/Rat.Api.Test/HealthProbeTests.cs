using System.Net.Http;
using System.Text.Json;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc.Testing;
using Rat.Api.Observability.Health.Responses;
using Snapshooter.Xunit;
using Xunit;

namespace Rat.Api.Test
{
    public class HealthProbeTests : IClassFixture<CustomWebApplicationFactory<Startup>>
    {
        private readonly HttpClient _client;

        public HealthProbeTests(CustomWebApplicationFactory<Startup> factory)
        {
            _client = factory.CreateClient(new WebApplicationFactoryClientOptions
            {
                AllowAutoRedirect = false
            });
        }

        [Fact]
        public async Task Readiness_Should_Return_Unhealthy_Response()
        {
            var healthResponse = await _client.GetAsync("/health/ready");

            using (var stream = await healthResponse.Content.ReadAsStreamAsync())
            {
                var healthCheckResponse = await JsonSerializer.DeserializeAsync<HealthCheckResponse>(stream);

                Snapshot.Match(
                    healthCheckResponse,
                    options => options.IgnoreField<double>("HealthChecks[*].Report.ElapsedMilliseconds"));
            }
        }
    }
}
