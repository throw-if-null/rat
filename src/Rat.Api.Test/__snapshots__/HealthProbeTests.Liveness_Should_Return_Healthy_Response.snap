{
  "Status": "Unhealthy",
  "HealthChecks": [
    {
      "Name": "Liveness probe",
      "Report": {
        "Status": "Unhealthy",
        "Description": "Dead",
        "ElapsedMilliseconds": 1.2159,
        "Tags": [
          "live"
        ],
        "Data": {}
      }
    }
  ]
}
